src/app/api/services/paginator/paginator.service.ts

```diff
@@ -0,0 +1,203 @@
+import { CoreWebService } from '../core-web-service';
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs/Observable';
+import { RequestMethod, URLSearchParams } from '@angular/http';
+
+/**
+ * Provides util listing methods
+ * @export
+ * @class PaginatorService
+ */
+@Injectable()
+export class PaginatorService {
+
+    public static readonly LINK_HEADER_NAME = 'Link';
+    public static readonly PAGINATION_PER_PAGE_HEADER_NAME = 'X-Pagination-Per-Page';
+    public static readonly PAGINATION_CURRENT_PAGE_HEADER_NAME = 'X-Pagination-Current-Page';
+    public static readonly PAGINATION_MAX_LINK_PAGES_HEADER_NAME = 'X-Pagination-Link-Pages';
+    public static readonly PAGINATION_TOTAL_ENTRIES_HEADER_NAME = 'X-Pagination-Total-Entries';
+
+   public static readonly OrderDirection = {
+        ASC: 'ASC',
+        DESC: 'DESC'
+    };
+
+    public links: Links = {};
+
+    public paginationPerPage: number;
+    public currentPage: number;
+    public maxLinksPage: number;
+    public totalRecords: number;
+
+    private _url: string;
```

Comment:
this is a standard in ts, you can see a example here:

https://github.com/Microsoft/TypeScript-Handbook/blob/3436aca106c664ee77691642c14b295c148cfc75/pages/Classes.md